File: problems/1716_calculate_money_in_leetcode_bank/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int totalMoney(int n) {
        int weeks = n / 7, days = n % 7, total = 0;
        for (int w = 0; w < weeks; w++) total += 28 + 7 * w;
        for (int d = 0; d < days; d++) total += weeks + 1 + d;
        return total;
    }
};

File: problems/1716_calculate_money_in_leetcode_bank/solution.py
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks = n // 7
        days = n % 7
        total = 0
        for w in range(weeks):
            total += 28 + 7 * w
        for d in range(days):
            total += weeks + 1 + d
        return total
